app/bot/__init__.py

```diff
@@ -110,58 +110,50 @@ async def on_ready():
 async def on_message(message: discord.Message):
     # Don't respond to messages from different guilds if it is enforced
     if not _in_guild(message):
-        if _botconf.botconfig.enforce_guild:
+        if _botconf.bot_config.enforce_guild:
             return
 
     # Don't respond to this bot's own messages
     if message.author == client.user:
         return
 
-    if _is_greeting(message):
-        logger.info("received greeting")
-        greeting = random.choice(_botconf.botconfig.greetings)
-        logger.info(f"greeting: {greeting}")
-        response = await llm.generate_response(
-            message,
-            _botconf.botconfig.system_prompt +
-            " You will respond with a somewhat short greeting and mention their name." +
-            f" You will incorperate the phrase \"{greeting}\" into your greeting." +
-            f" Their name is {message.author.name}.",
-        )
-        logger.info(f"response: `{response}`")
-        if response is not None:
-            await message.reply(response)
-            await client.change_presence(status=discord.Status.online)
-        else:
-            await client.change_presence(status=discord.Status.idle)
-        return
-
     if _is_command(message.content):
         logger.info("received command")
         split_message = message.content.strip(" \t\n").split()
-        command = split_message[0][len(_botconf.botconfig.command_prefix):]
+        command = split_message[0][len(_botconf.bot_config.command_prefix):]
         args = split_message[1:]
 
         await _handle_command(command, args, message)
         return
 
+    # Add message to history
+    _bothist.bot_history.add_message(message)
```

Comment:
One issue that I realized on Ruby Bot, is that this chat history isn't thread safe, so there might be issues where the history gets a little out of order if it receives multiple messages while still processing an existing one. 

There isn't really a great solution to that though, other than maybe appending both user and bot responses only after the response is generated, but you'd need a way to temporarily add the current message so it knows to generate the current response.

Reply:
One possible way is to also include a timestamp along with the role and content, and then insert the message into where it would go chronologically instead of just appending it